### PR TITLE
[ADD] account_tax_group_widget_base_amount: New module

### PR DESCRIPTION
**Pull request by chienandalu**
This module adds the base value to the tax group widget used in invoices and receipts
as it's shown in reports.

image

cc @Tecnativa TT29538

ping @pedrobaeza @victoralmau

### CODE REVIEW

File: account_tax_group_widget_base_amount/README.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+.. contents::
+   :local:
+
+Known issues / Roadmap
+======================
+
+* With the refactor of the tax group widget in v15, hopefully this module won't be
+  necessary. As for the moment that change is still in developement, check when the
+  moment to migrate to such version arrives.
```
Comment (pedrobaeza) [inline account_tax_group_widget_base_amount/README.rst:44]: Mention here the PR link for reference

File: account_tax_group_widget_base_amount/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+This module adds the base value to the tax group widget used in invoices and receipts
```
Comment (pedrobaeza) [inline account_tax_group_widget_base_amount/readme/DESCRIPTION.rst]:
```suggestion
This module adds the base amount to the tax group widget used in invoices and receipts
```

File: account_tax_group_widget_base_amount/static/src/xml/tax_group.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates>
+    <t t-extend="AccountTaxGroupTemplate">
+        <t t-jquery=".oe_tax_group_name" t-operation="replace">
```
Comment (pedrobaeza) [inline account_tax_group_widget_base_amount/static/src/xml/tax_group.xml]: Can you avoid a `replace` operation?

File: account_tax_group_widget_base_amount/README.rst (hunk 1 above)
Comment (chienandalu) [reply account_tax_group_widget_base_amount/README.rst:44]: done

File: account_tax_group_widget_base_amount/readme/DESCRIPTION.rst (hunk 2 above)
Comment (chienandalu) [reply account_tax_group_widget_base_amount/readme/DESCRIPTION.rst]: done

File: account_tax_group_widget_base_amount/static/src/xml/tax_group.xml (hunk 3 above)
Comment (chienandalu) [reply account_tax_group_widget_base_amount/static/src/xml/tax_group.xml]: Done. I wasn't sure I could use a t-jquery select to get t tags

**Review by victoralmau — APPROVED** [review]
Code review.

File: account_tax_group_widget_base_amount/static/src/xml/tax_group.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates>
+    <t t-extend="AccountTaxGroupTemplate">
+        <t t-jquery="t[t-esc='line[0]']" t-operation="after">
```
Comment (pedrobaeza) [inline account_tax_group_widget_base_amount/static/src/xml/tax_group.xml:4]: For clarity, you should only show this if the base is different from the total base. See how is done in the report.

Comment (chienandalu) [reply account_tax_group_widget_base_amount/static/src/xml/tax_group.xml:4]: Ok, it needed some js override, but now we show it only when there are multiple tax groups

Comment (pedrobaeza) [reply account_tax_group_widget_base_amount/static/src/xml/tax_group.xml:4]: Why not using at least `len(lines) > 1` for not overriding JS?

File: account_tax_group_widget_base_amount/static/src/js/tax_group.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            return this.value && this.value.length > 1;
+        },
+        _render: function() {
+            var _this = this;
+            if (this._isMultiTax()) {
+                this.$el.html(
+                    $(
+                        QWeb.render("AccountTaxGroupTemplate", {
+                            lines: _this.value,
+                            displayBase: true,
```
Comment (pedrobaeza) [inline account_tax_group_widget_base_amount/static/src/js/tax_group.js]: What about `displayEditWidget: displayEditWidget,` that I found in core?

File: account_tax_group_widget_base_amount/static/src/xml/tax_group.xml (hunk 4 above)
Comment (chienandalu) [reply account_tax_group_widget_base_amount/static/src/xml/tax_group.xml:4]: Sure. Dumb of me. Done

File: account_tax_group_widget_base_amount/static/src/js/tax_group.js (hunk 5 above)
Comment (chienandalu) [reply account_tax_group_widget_base_amount/static/src/js/tax_group.js]: I've redone it to avoid js override

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**